src/core/tests/tensor.cpp

```diff
@@ -88,5 +91,116 @@ TEST(tensor, get_byte_size_u6_not_even_div_by_storage_unit) {
+    static constexpr ov::element::Type ov_type = ov::element::from<element_type>();
+
+    void SetUp() override {
+        shape = {1, 2, 3, 4};
+        auto static_shape = shape.get_shape();
+        size_t elements = std::accumulate(static_shape.begin(), static_shape.end(), 1, std::multiplies<size_t>());
+        data.resize(elements);
+        element_type next_value{};
+        std::generate(data.begin(), data.end(), [&next_value]() {
+            next_value += 1;
+            return next_value;
+        });
+
+        file_name_str = ov::test::utils::generateTestFilePrefix();
+        file_name = path_type(file_name_str.begin(), file_name_str.end());
+    }
+
+    void remove_file() {
+        if (std::filesystem::exists(file_name))
+            std::filesystem::remove(file_name);
+    }
+
+    ov::PartialShape shape;
+    std::vector<element_type> data;
+    path_type file_name;
+    std::string file_name_str;
+};
+
+TYPED_TEST_SUITE_P(OffloadedTensorTest);
+
+TYPED_TEST_P(OffloadedTensorTest, save_tensor) {
+    using element_type = typename std::remove_reference_t<decltype(*this)>::element_type;
+    ov::Tensor tensor(this->ov_type, this->shape.get_shape());
+    size_t data_size = this->data.size() * sizeof(element_type);
+    std::memcpy(tensor.data(), this->data.data(), data_size);
+
+    EXPECT_NO_THROW(save_tensor_data(tensor, this->file_name));
+    ASSERT_TRUE(std::filesystem::exists(this->file_name));
+    {
+        std::ifstream fin(this->file_name_str, std::ios::binary);
+        std::vector<element_type> file_data(this->data.size());
+        fin.read(reinterpret_cast<char*>(file_data.data()), data_size);
+        EXPECT_TRUE(std::equal(this->data.begin(), this->data.end(), file_data.begin()));
+    }
+    this->remove_file();
+}
+
+TYPED_TEST_P(OffloadedTensorTest, read_tensor) {
+    using element_type = typename std::remove_reference_t<decltype(*this)>::element_type;
+
+    size_t data_size = this->data.size() * sizeof(element_type);
+    {
+        std::ofstream fout(this->file_name_str, std::ios::binary);
+        fout.write(reinterpret_cast<char*>(this->data.data()), data_size);
+    }
+    ASSERT_TRUE(std::filesystem::exists(this->file_name));
+
+    ov::Tensor tensor;
+    EXPECT_NO_THROW(tensor = read_tensor_data(this->file_name, this->ov_type, this->shape, 0, true));
+
+    {
+        std::vector<element_type> file_data(this->data.size());
+        std::memcpy(file_data.data(), tensor.data(), data_size);
+
+        EXPECT_TRUE(std::equal(this->data.begin(), this->data.end(), file_data.begin()));
+    }
+    this->remove_file();
+}
+
+TYPED_TEST_P(OffloadedTensorTest, create_mmaped_tensor) {
+    using element_type = typename std::remove_reference_t<decltype(*this)>::element_type;
+    ov::Tensor tensor(this->ov_type, this->shape.get_shape());
+    size_t data_size = this->data.size() * sizeof(element_type);
+    std::memcpy(tensor.data(), this->data.data(), data_size);
```

Comment:
(if I'm not mixing type domains) It may fail for _under byte_ types. [Look here](https://github.com/openvinotoolkit/openvino/blob/58b02fb88fa660468d281e751f117c51cb22932c/src/core/src/descriptor/tensor.cpp#L218)

Reply:
I changed it to test also such types